packages/shared/components/modals/AccountActions.svelte

```diff
@@ -16,41 +16,43 @@
 
     export let isActive
 
-    const account = getContext<Readable<WalletAccount>>('selectedAccount')
     const viewableAccounts = getContext<Readable<WalletAccount[]>>('viewableAccounts')
     const allAccounts = getContext<Readable<WalletAccount[]>>('walletAccounts')
     const hiddenAccounts = $activeProfile?.hiddenAccounts ?? []
 
     const hidden = hiddenAccounts.includes($selectedAccountId)
     const canDelete =
-        $account.index === $allAccounts.length - 1 && $account.rawIotaBalance === 0 && $account.messages.length === 0
+        $selectedAccount.index === $allAccounts.length - 1 &&
+        $selectedAccount.rawIotaBalance === 0 &&
+        $selectedAccount.messages.length === 0
 
     const handleCustomiseAccountClick = () => {
         accountRoute.set(AccountRoutes.Manage)
         isActive = false
     }
 
     const handleViewAddressHistoryClick = () => {
-        openPopup({ type: 'addressHistory', props: { account } })
+        openPopup({ type: 'addressHistory', props: { selectedAccount } })
```

Comment:
This part causes a crash for me. I think you need to adjust this in AddressHistory, since it's expecting a readable store, but selectedAccount is writable. Imo, we should expect a `WalletAccount` there instead of a store and then pass the value instead of the store. Also the prop should be named, since the variable is called differently withing AddressHistory.

Reply:
fixed, i was using the wrong prop

Reply:
@Tuditi  About the other changes to improve the code, I agree, but I wouldnt add that here. We should add these suggestions to the milestone to improve code and just work on these little by little. Could you please add your suggestions [here](https://github.com/iotaledger/firefly/milestone/1) 🙏🏼 ?

Reply:
Done! https://github.com/iotaledger/firefly/issues/2396